src/main/java/com/intuit/graphql/orchestrator/schema/transform/KeyTransformer.java

```diff
@@ -24,20 +28,24 @@
 
   @Override
   public XtextGraph transform(final XtextGraph source) {
-    List<TypeDefinition> entities = source.getTypes().values().stream()
+    Map<String, TypeDefinition> entities = source.getTypes().values().stream()
             .filter(typeDefinition -> typeContainsDirective(typeDefinition, FEDERATION_KEY_DIRECTIVE))
```

Comment:
I tried using the union entity, but it would take longer. If we used the union entity, we would then have to filter to get the type definition(for the validation portion) which would require it to process the types one time per entity instead of just collecting filtering the types once. If we make the parser add the type definition information for the _entites, thenn that would be the better option.